lib/kb_fasttree/fasttree.py

```diff
@@ -0,0 +1,738 @@
+"""Implementation of the fasttree app."""
+
+import os
+import re
+import subprocess
+import sys
+from datetime import datetime, timezone
+from pathlib import Path
+from pprint import pformat
+from typing import Any, Dict
+
+import ete3  # Tree drawing package we install in Dockerfile
+from installed_clients.DataFileUtilClient import DataFileUtil
+from installed_clients.KBaseReportClient import KBaseReport
+from installed_clients.WorkspaceClient import Workspace
+
+SERVICE_VER = "release"
+FASTTREE_BIN = "/kb/module/FastTree/bin/FastTree"
+
+
+def generate_run_id() -> str:
+    """Generate an ID for this fasttree run, based on the current date and time.
+
+    :return: temporally unique run ID
+    :rtype: str
+    """
+    return datetime.now(tz=timezone.utc).strftime("%Y%m%d_%H%M%S")
```

Comment:
new: previously this app produced files with either a UUID or a timestamp as part of the filename. Instead of doing that, the app now creates a directory (in the `scratch` dir) with the timestamp as its name, and adds the files _sans_ UUID/timestamp to that directory. It means that app generates a predictable set of files, which makes testing easier, and if for some reason you run the app numerous times in a row, each run will have its output in a separate directory.